### PR TITLE
Adding additional default use cases

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
- Adding two additional default use cases for pretrained model for hybrid and semantic search
- removing `id` field from some of the use case templates
- removing weights from hybrid search template
- making the text and image type dynamic for multimodal search as most customers don't actually just use binary for images. 
- Adding e2e integration test for local semantic search template all the way to neural search

### Issues Resolved
part of #666 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by joshpalis** [comment]
Interesting that the windows single node integration tests are failing due to : 
```
EPRODUCE WITH: gradlew ':integTest' --tests "org.opensearch.flowframework.rest.FlowFrameworkRestApiIT.testSemanticSearchWithLocalModelEndToEnd" -Dtests.seed=F9C1F84B83F9D1C6 -Dtests.security.manager=false -Dtests.locale=mk -Dtests.timezone=PLT -Druntime.java=11

org.opensearch.flowframework.rest.FlowFrameworkRestApiIT > testSemanticSearchWithLocalModelEndToEnd FAILED
    java.lang.AssertionError: expected:<0> but was:<1>
Suite: Test class org.opensearch.flowframework.rest.FlowFrameworkRestApiIT
        at __randomizedtesting.SeedInfo.seed([F9C1F84B83F9D1C6:9469A614D9123AA7]:0)
  2> SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".  2> 
        at org.junit.Assert.fail(Assert.java:89)
```
Looking at the test, its failing during this line : `assertEquals(neuralSearchResponse.getHits().getHits().length, 1);`
Given that the other integration tests passed, perhaps this failure is due to the GHA runner memory limitation. I see that the model used in the semantic search local model template is `msmarco-distilbert-base-tas-b`, which is about 254 mbs. 

Can we use a smaller model here for this test?

File: build.gradle [hunk 1]
```diff
@@ -181,6 +181,8 @@ dependencies {
 
     // ZipArchive dependencies used for integration tests
     zipArchive group: 'org.opensearch.plugin', name:'opensearch-ml-plugin', version: "${opensearch_build}"
+    zipArchive group: 'org.opensearch.plugin', name:'opensearch-knn', version: "${opensearch_build}"
+    zipArchive group: 'org.opensearch.plugin', name:'neural-search', version: "${opensearch_build}"
```
Comment (owaiskazi19) [inline build.gradle:185]: Do we want to commit this?

Comment (amitgalitz) [reply build.gradle:185]: Yes it was added for IT testing, if I can't get BWC to work though I will remove this and the IT I added and add that in a different PR.

File: src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java [hunk 2]
```diff
@@ -95,10 +95,20 @@ public PlainActionFuture<WorkflowData> execute(
...
-            Integer slices = (Integer) inputs.get(SLICES);
-            Integer maxDocs = (Integer) inputs.get(MAX_DOCS);
-
+            Integer slices;
+            Integer maxDocs;
+            if (inputs.get(SLICES) != null) {
+                slices = Integer.parseInt(String.valueOf(inputs.get(SLICES)));
+            } else {
+                slices = (Integer) inputs.get(SLICES);
+            }
```
Comment (owaiskazi19) [inline src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java:106]:
```java
slices = inputs.get(SLICES) instanceof String
         ? Integer.parseInt(String.valueOf(inputs.get(SLICES)))
         : (Integer) inputs.get(SLICES);
```

Comment (amitgalitz) [reply src/main/java/org/opensearch/flowframework/workflow/ReindexStep.java:106]: I need to a != null check and with ternary operator you can't do that cause you need both sides of : to not be null unlike if/else

**Review by ohltyler — APPROVED** [review]
At a high level LGTM.

**Review by joshpalis — APPROVED** [review]

**Pull request merged**